pytest_django/plugin.py

```diff
@@ -514,6 +514,20 @@ def _template_string_if_invalid_marker(request):
             else:
                 dj_settings.TEMPLATE_STRING_IF_INVALID.fail = False
 
+
+@pytest.fixture(autouse=True, scope='function')
+def django_clear_site_cache():
+    """Clears ``django.contrib.sites.models.SITE_CACHE`` to avoid
+    unexpected behavior when cached site objects.
+    """
+
+    if django_settings_is_configured():
+        from django.conf import settings as dj_settings
+
+        if 'django.contrib.sites' in dj_settings.INSTALLED_APPS:
+            from django.contrib.sites import models
+            models.SITE_CACHE.clear()
```

Comment:
I think this PR is reasonable. People that are not using the site framework (not having it in INSTALLED_APPS) won't notice any difference. 

There is a helper to clear the cache `Site.objects.clear_cache()` that I think should be used instead of directly clearing the SITE_CACHE which from what I understand is not a public API:
https://docs.djangoproject.com/en/1.10/ref/contrib/sites/#caching-the-current-site-object